curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/6555de565387a2efe90a6ccc.md

```diff
@@ -34,7 +34,7 @@ assert.match(code, /playButton\.setAttribute\(\s*('|")aria-label\1/)
 Your `setAttribute` method should have ``song?.title ? `Play ${song.title}` : "Play"`` as the second argument.
 
 ```js
-assert.match(code, /playButton\.setAttribute\(\s*('|")aria-label\1,\s*song\?\.title\s*\?\s*`Play\s*\$\{song\.title\}`\s*:\s*\1Play\1\s*\);?\s*/)
+assert.match(code, /playButton\.setAttribute\(\s*('|")aria-label\1,\s*song\?\.title\s*\?\s*`Play\s*\$\{song\.title\}`\s*:\s*('|")Play\1\s*\);?\s*/)
```

Comment:
The back-reference after `Play` (i.e `\1`) refers back to the first quote type. This would inappropriately allow `setAttribute("aria-label", … 'play")` to pass. I think instead the back reference needs to be `\2`. But please test and confirm.

Reply:
Oh right. I agree. There should be a `\2` instead of `\1`. Otherwise, it will allow inappropriate cases like the one you mentioned pass. I tested `\2` and it works fine. 

#### Cases - 
1. Different quotes

![image](https://github.com/freeCodeCamp/freeCodeCamp/assets/82658685/89dc2c25-a25e-406c-b7d7-2875f6c56dc4)
![image](https://github.com/freeCodeCamp/freeCodeCamp/assets/82658685/3e68a842-d319-4eb3-bab1-cd8878a8d571)

2. Same quotes

![image](https://github.com/freeCodeCamp/freeCodeCamp/assets/82658685/87040731-50c6-4111-aafc-2a4a38803e33)

##### Failing case - 

![image](https://github.com/freeCodeCamp/freeCodeCamp/assets/82658685/834f1ba8-863c-4d27-8655-e8eaf70f5285)